src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,147 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev,T value, Node<T> next) {
+            
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (first == null) {
+            first = last = newNode;
+        } else {
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index, size + 1);
+        Node<T> newNode = new Node<>(null, value, first);
+        if (first == null) {
+            first = last = newNode;
+        } else if (size == 0) {
+            add(value);
```

Comment:
```suggestion
         if (size == 0) {
            add(value);
            return;
         }
```
size == null і first == null це по суті одне й те саме (пустий ліст)